distribution/pull_v2.go

```diff
@@ -94,12 +95,24 @@ func (p *v2Puller) Pull(ctx context.Context, ref reference.Named) (err error) {
 
 func (p *v2Puller) checkTrusted(ref reference.Named) (reference.Named, error) {
 	p.originalRef = ref
-	imgRef, err := docker.NewReference(ref)
+	dockerRef, err := containersImageRef.ParseNamed(ref.String())
```

Comment:
ACK. It is a bit horrible, but that’s what we get with Golang’s interface-as-collection-of-method-signatures type system instead of an interface-as-a-name-and-explicitly-committed-to-protocol.

Please make sure to add a comment explaining why we can't use `docker.NewReference` without the `ParseNamed(ref.String())` roundtrip so that this isn’t “cleaned up” later when we forget or a new maintainer comes.